docs/containers-registries.conf.5.md

```diff
@@ -18,80 +18,141 @@ VERSION 2 is the latest format of the `registries.conf` and is currently in
-specified; this is a list of host[:port] (not more general prefix!) entries to
-use when pulling an unqualified image, in order.
+### GLOBAL SETTINGS
+
+`unqualified-search-registries`
+: An array of host[:port] (not more general prefix!) entries to use when pulling
+an unqualified image, in order.
+
+### NAMESPACED `[[registry]]` SETTINGS
+
+The following settings may differ among different registries, and namespaces/repositories
+within a registry.  The configuration is represented as a sequence of `[[registry]]`
+entries containging the specific settings.
+
+#### Choosing a `[[registry]]` entry
+
+A `[[registry]]` entry is chosen based on its `prefix` field; only one `[[registry]]`
+entry, the one with the best-matching `prefix`, is used.
+
+`prefix`
+: A prefix of the user-specified image name, i.e. using one of the following formats:
+    - _host_[`:`_port_]
+    - _host_[`:`_port_]`/`_namespace_[`/`_namespace_…]
+    - _host_[`:`_port_]`/`_namespace_[`/`_namespace_…]`/`_repo_
+    - _host_[`:`_port_]`/`_namespace_[`/`_namespace_…]`/`_repo_(`:`_tag|`@`_digest_)
+
+    The user-specified image name must start with the specified `prefix` for the
+    `[[registry]]` entry to be considered (and continue with the appropriate separator),
+    and (only) the entry with the longest match is used.
+
+    As a special case, the `prefix` field can be missing; if so, it defaults to the value
+    of the `location` field (described below).
+
+#### Per-namespace settings
+
+`insecure`
+: `true` or `false`.
+    By default, container runtimes require TLS when retrieving images from a registry.
+    If `insecure` is set to `true`, unencrypted HTTP as well as TLS connections with untrusted
+    certificates are allowed.
+
+`blocked`
+: `true` or `false`.
+    If `true`, pulling images with matching names is forbidden.
+
+#### Remapping and mirroring registries
+
+The user-specified image reference is, primarily, a "logical" image name, always used for naming
+the image.  By default, the image reference also directly specifies there registry and repository
+to use, but the following options can be used to redirect the undeflying accesses
+to different registry servers or locations (e.g. to support configurations with no access to the
+internet without having to change `Dockerfile`s, or to add redundancy).
+
+`location`
+: Accepts the same format as the `prefix` field, and specifies the physical location
+    of the `prefix`-rooted namespace.
+
+    By default, this equal to `prefix` (in which case `prefix` can be omitted and the
+    `[[registry]]` entry can only specify `location`).
+
+    Example: Given
+    ```
+    prefix = "example.com/foo"
+    location = "internal-registry-for-example.net/bar"
+    ```
+    requests for the image `example.com/foo/myimage:latest` will actually work with the
+    `internal-registry-for-example.net/bar/myimage:latest` image.
+
+`mirror`
+: An array of TOML entries specifiying (possibly-partial) mirrors for the
```

Comment:
I have (finally) read the TOML spec (“obvious, minimal”… not), and updated the text throughout to say array / TOML table.  (“TOML table” instead of just “table” because the underlying meaning of “hash table” is implementation-focused and completely divorced from the general meaning of the “table” word).